controllers/mattermost/mattermost/mattermost.go

```diff
@@ -373,31 +395,35 @@ func (r *MattermostReconciler) checkUpdateJob(
 		job.Spec.Template.Spec.Containers,
 	)
 	if err != nil {
-		return nil, errors.Wrap(err, "failed to compare image of update job and desired deployment")
+		return nil, reconcileStatus{}, errors.Wrap(err, "failed to compare image of update job and desired deployment")
 	}
 	if !isSameImage {
 		reqLogger.Info("Mattermost image changed, restarting update job")
-		err := r.Resources.RestartMattermostUpdateJob(mattermost, job, baseDeployment, reqLogger)
+		err = r.Resources.RestartMattermostUpdateJob(mattermost, job, baseDeployment, reqLogger)
 		if err != nil {
-			return nil, errors.Wrap(err, "failed to restart update job")
+			recStatus.ResourcesReady = false
+			return nil, recStatus, errors.Wrap(err, "failed to restart update job")
 		}
 
-		return nil, errors.New("Restarted update image job")
+		recStatus.ResourcesReady = false
+		return nil, recStatus, errors.New("Restarted update image job")
```

Comment:
We restarted the job successfully, so again we just want `ResourcesReady == false` and no error.

```suggestion
		return nil, recStatus, nil
```